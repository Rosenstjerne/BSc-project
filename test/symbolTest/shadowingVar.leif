var int global_var;

function int shadow_test() {
    var int global_var = 10;
    return global_var;
}

global_var = 5;

print(global_var); # Should print 5
print(shadow_test()); # Should print 10
print(global_var); # Should print 5
