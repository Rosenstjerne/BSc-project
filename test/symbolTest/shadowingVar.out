function null main() {
	var int global_var;
	
	function int shadow_test() {
		var int global_var;
		
		global_var = 10;
		return global_var;
		
	}
	
	global_var = 5;
	print(global_var);
	print(shadow_test());
	print(global_var);
	
}
